python_onnxifi/data_conversion.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pytypes.h>
+
+#include "onnx/onnxifi.h"
+#include "onnx/onnx.pb.h"
+#include "onnx/shape_inference/implementation.h"
+
+#include <vector>
+
+namespace py = pybind11;
+
+using ::ONNX_NAMESPACE::ModelProto;
+// Utility class to navigate conversions of onnxTensorDescriptor and numpy
+// arrays
+// NOTE: py::dict is not const anywhere because pybind11 request function is not
+
+// Structor to manage onnxTensorDescriptor data for lifetime of the descriptor
+struct DescriptorData {
+  DescriptorData();
+  DescriptorData(const DescriptorData& d);
+  // Ensure pointers point to beginning of vectors/strings
+  DescriptorData(DescriptorData&& d) noexcept;
+
+  onnxTensorDescriptor descriptor;
+  std::vector<uint64_t> shape;
+  std::string name;
+  // container for bytes
+  std::vector<char> buffer;
```

Comment:
What's the usage of this one? Data is already stored in `descriptor.buffer` right?

Reply:
Yup, but the vector is for memory management

Reply:
OK. I see.